src/aosm/azext_aosm/delete/delete.py

```diff
@@ -0,0 +1,330 @@
+
+        if isinstance(self.config, VNFConfiguration):
+            self.delete_artifact_manifest("sa")
+        self.delete_artifact_manifest("acr")
+
+        if clean:
+            logger.info("Delete called for all resources.")
+            self.delete_nfdg()
+            self.delete_artifact_store("acr")
+            if isinstance(self.config, VNFConfiguration):
+                self.delete_artifact_store("sa")
+            self.delete_publisher()
+
+    def delete_nsd(self, clean: bool = False, force: bool = False) -> None:
+        """
+        Delete the NSDV and manifests.
+
+        If they don't exist it still reports them as deleted.
+        """
+        assert isinstance(self.config, NSConfiguration)
+
+        if not force:
+            print(
+                "Are you sure you want to delete the NSD Version"
+                f" {self.config.nsd_version}, the associated manifests"
+                f" {self.config.acr_manifest_names} and configuration group schema"
+                f" {self.config.cg_schema_name}?"
+            )
+            if clean:
+                print(
+                    f"Because of the --clean flag, the NSD {self.config.nsd_name} will also be deleted."
+                )
+            print("There is no undo. Type 'delete' to confirm")
+            if not input_ack("delete", "Confirm delete:"):
+                print("Not proceeding with delete")
+                return
+
+        self.delete_nsdv()
+        self.delete_artifact_manifest("acr")
+        self.delete_config_group_schema()
+        if clean:
+            self.delete_nsdg()
+
+    def delete_nfdv(self):
+        assert isinstance(self.config, NFConfiguration)
+        message = (
+            f"Delete NFDV {self.config.version} from group {self.config.nfdg_name} and"
+            f" publisher {self.config.publisher_name}"
+        )
+        logger.debug(message)
+        print(message)
+        try:
+            poller = self.api_clients.aosm_client.network_function_definition_versions.begin_delete(
+                resource_group_name=self.config.publisher_resource_group_name,
+                publisher_name=self.config.publisher_name,
+                network_function_definition_group_name=self.config.nfdg_name,
+                network_function_definition_version_name=self.config.version,
+            )
+            LongRunningOperation(self.cli_ctx, "Deleting NFDV...")(poller)
+            print("Deleted NFDV.")
+        except Exception:
+            logger.error(
+                "Failed to delete NFDV %s from group %s",
+                self.config.version,
+                self.config.nfdg_name,
+            )
+            raise
+
+    def delete_nsdv(self):
+        assert isinstance(self.config, NSConfiguration)
+        message = (
+            f"Delete NSDV {self.config.nsd_version} from group"
+            f" {self.config.nsd_name} and publisher {self.config.publisher_name}"
+        )
+        logger.debug(message)
```

Comment:
Why print and record debug logs simultaneously?

Reply:
We need to rethink how we output to console. These changes are too widespread to make before GA release, so I've opened work item https://dev.azure.com/msazuredev/AzureForOperators/_workitems/edit/992775 to track.